source/elements/oneDAL/source/data_management/data_sources.rst

```diff
@@ -1,7 +1,117 @@
+.. highlight:: cpp
+.. default-domain:: cpp
+
 .. _data-sources:
 
 ============
 Data Sources
 ============
 
-TBD
+This section describes the types related to the :txtref:`data-source` concept.
+
+---------------
+CSV data source
+---------------
+Class ``csv::data_source`` is an api for accessing the data source represented as a :capterm:`csv file <CSV file>`.
+CSV data source can be used in :expr:`read` operation to extract data in text format from the given input file,
+process it using provided parameters (such as delimiter and parsing options), transform it into numerical representation,
+and store it as an in-memory :txtref:`dataset` of a chosen type.
+
+Supported type of in-memory dataset for :expr:`read` operation with CSV data source is :txtref:`table`.
+
+CSV data source requires input file name to be set in the constructor, while other parameters can be initialized with default values.
+
+-------------
+Usage example
+-------------
+::
+
+   const auto data_source = onedal::csv::data_source("data.csv", ',');
+
+   const auto table = ondedal::read<onedal::table>(data_source);
+
+
+---------------------
+Programming Interface
+---------------------
+All type and function in this section shall be declared in the
+``oneapi::dal::csv`` namespace and be available via inclusion of the
+``oneapi/dal/io/csv.hpp`` header file.
+
+::
+
+   enum class read_options {
```

Comment:
do we want to explicitly specify the type used to store the data such as uint32_t or different as the base class?

Reply:
Done.